shared/output/gapic/templates/grpc_service_config/lib/testing/grpc_service_config/service_with_retries/client.rb

```diff
@@ -0,0 +1,282 @@
+        #   @param request [Testing::GrpcServiceConfig::Request | Hash]
+        #   @param options [Gapic::CallOptions, Hash]
+        #     Overrides the default settings for this call, e.g, timeout, retries, etc. Optional.
+        #
+        #
+        # @yield [response, operation] Access the result along with the RPC operation
+        # @yieldparam response [Testing::GrpcServiceConfig::Response]
+        # @yieldparam operation [GRPC::ActiveCall::Operation]
+        #
+        # @return [Testing::GrpcServiceConfig::Response]
+        #
+        # @raise [GRPC::BadStatus] if the RPC is aborted.
+        #
+        def method_level_retry_method request, options = nil
+          raise ArgumentError, "request must be provided" if request.nil?
+
+          request = Gapic::Protobuf.coerce request, to: Testing::GrpcServiceConfig::Request
+
+          # Converts hash and nil to an options object
+          options = Gapic::CallOptions.new options.to_h if options.respond_to? :to_h
+
+          # Customize the options with defaults
+          metadata = @config.rpcs.method_level_retry_method.metadata.to_h
+
+          # Set x-goog-api-client header
+          metadata[:"x-goog-api-client"] ||= Gapic::Headers.x_goog_api_client \
+            lib_name: @config.lib_name, lib_version: @config.lib_version,
+            gapic_version: Testing::GrpcServiceConfig::VERSION
+
+          options.apply_defaults timeout:      @config.rpcs.method_level_retry_method.timeout,
+                                 metadata:     metadata,
+                                 retry_policy: @config.rpcs.method_level_retry_method.retry_policy
+          options.apply_defaults metadata:     @config.metadata,
+                                 retry_policy: @config.retry_policy
+
+          @service_with_retries_stub.call_rpc :method_level_retry_method, request, options: options do |response, operation|
+            yield response, operation if block_given?
+            return response
+          end
+        end
+
+        ##
+        # Configuration class for the ServiceWithRetries API.
+        class Configuration
+          extend Gapic::Config
+
+          config_attr :endpoint,     "grpcserviceconfig.example.com", String
+          config_attr :credentials,  nil do |value|
+            allowed = [::String, ::Hash, ::Proc, ::Google::Auth::Credentials, ::Signet::OAuth2::Client, nil]
+            allowed += [::GRPC::Core::Channel, ::GRPC::Core::ChannelCredentials] if defined? ::GRPC
+            allowed.any? { |klass| klass === value }
+          end
+          config_attr :scope,        nil, String, Array, nil
+          config_attr :lib_name,     nil, String, nil
+          config_attr :lib_version,  nil, String, nil
+          config_attr(:channel_args, { "grpc.service_config_disable_resolution"=>1 }, Hash, nil)
+          config_attr :interceptors, nil, Array, nil
+          config_attr :timeout,      nil, Numeric, nil
```

Comment:
Why not set the default `timeout` config value of `20` on this line? Right now it's defined in the second method argument with a default value of `nil`, but then is overwritten on line 56. Why not set the value as the default?

Reply:
@blowmage We don't want it set in _every_ instance of the config, only the default instance. This is so that users can modify the default instance, and those modifications won't get overridden by the child config when a client is instantiated.

Reply:
Isn’t the way it’s written that a config object will look to a parent config object when one is set? In that case then it shouldn’t matter if a child config object has the same defaults or not. It will still defer to the parent object, right?

Reply:
@blowmage Ah, you're right, I was mistaken—I thought it checked the default first and then the parent object second. But of course that wouldn't make sense. It does check the parent object first.

So yes, setting the default would work for the service-wide fields. However, it still wouldn't work for per-method configs, since they all share the same class (`Gapic::Config::Method`) and thus the same `config_attr` defaults. So I still think it's better to set all of these on an object rather than use the defaults mechanism.

Reply:
Or you could create a per-RPC method config class with the correct defaults for that RPC. If there are no defaults for the RPC, then it can use `Gapic::Config::Method`:

```ruby
##
# Configuration RPC class for the ServiceWithRetries API.
class Rpcs
  class MethodLevelRetryMethod
    extend Gapic::Config

    config_attr :timeout,  60.0, Numeric, nil
    config_attr :metadata, nil, Hash, nil
    config_attr(
      :retry_policy, 
      { initial_delay: 1.0,
        max_delay:     10.0,
        multiplier:    3.0,
        retry_codes:   ["UNAVAILABLE"] }, 
      Hash, 
      Proc, 
      nil
    )

    ##
    # Create a new Config::Method object instance.
    #
    # @param parent_method [Gapic::Config::Method, nil] The config to look to values for.
    #
    def initialize parent_method = nil
      @parent_config = parent_method unless parent_method.nil?

      yield self if block_given?
    end
  end

  attr_reader :service_level_retry_method
  attr_reader :method_level_retry_method

  def initialize parent_rpcs = nil
    service_level_retry_method_config = parent_rpcs&.service_level_retry_method if parent_rpcs&.respond_to? :service_level_retry_method
    @service_level_retry_method = Gapic::Config::Method.new service_level_retry_method_config
    method_level_retry_method_config = parent_rpcs&.method_level_retry_method if parent_rpcs&.respond_to? :method_level_retry_method
    @method_level_retry_method = MethodLevelRetryMethod.new method_level_retry_method_config

    yield self if block_given?
  end
end
```

Reply:
Yeah we could. Still seems a lot simpler to set the default values in the base config object, than to be spreading it out across N new classes, though.